Source/WebCore/html/HTMLMediaElement.h

```diff
@@ -1081,19 +1081,15 @@ class HTMLMediaElement
 
     struct PendingSeek {
         WTF_MAKE_STRUCT_FAST_ALLOCATED;
-        PendingSeek(const MediaTime& now, const MediaTime& targetTime, const MediaTime& negativeTolerance, const MediaTime& positiveTolerance)
+        PendingSeek(const MediaTime& now, const SeekTarget& inTarget)
             : now(now)
-            , targetTime(targetTime)
-            , negativeTolerance(negativeTolerance)
-            , positiveTolerance(positiveTolerance)
+            , target(inTarget)
         {
         }
         MediaTime now;
-        MediaTime targetTime;
-        MediaTime negativeTolerance;
-        MediaTime positiveTolerance;
+        SeekTarget target;
     };
-    std::unique_ptr<PendingSeek> m_pendingSeek;
+    std::optional<PendingSeek> m_pendingSeek;
```

Comment:
We have done some work to move from optional to unique_ptr to reduce some object's memory footprint.
Do we need this change?

Reply:
I wasn't aware of this. I would have assumed here that allocation within the HTMMediaElement class is better than a single heap allocation with all the required overhead.